### PR TITLE
Implement beatmaps over IPC

### PR DESCRIPTION
**Pull request by ddevault**

### CODE REVIEW

**Comment by peppy** [comment]
I'd rather the osz file was included in the resource repository than this one, for licensing reasons. Also just to keep large files out of this repo.

**Comment by SirCmpwn** [comment]
Will move it.

File: osu.Game.Tests/osu.Game.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    <WarningLevel>4</WarningLevel>
+    <ConsolePause>false</ConsolePause>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="System" />
+    <Reference Include="nunit.framework">
+      <HintPath>..\packages\NUnit.2.6.4\lib\nunit.framework.dll</HintPath>
+    </Reference>
+    <Reference Include="OpenTK">
+      <HintPath>..\packages\ppy.OpenTK.1.1.2225.3\lib\net20\OpenTK.dll</HintPath>
```
Comment (SirCmpwn) [inline osu.Game.Tests/osu.Game.Tests.csproj]: :-1:

Comment (peppy) [inline osu.Game.Tests/osu.Game.Tests.csproj]: these are okay; it's only the ones in the framework projects that matter

File: osu.Game/Beatmaps/IO/ArchiveReader.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using osu.Framework.Platform;
+
+namespace osu.Game.Beatmaps.IO
+{
+    public abstract class ArchiveReader : IDisposable
```
Comment (peppy) [inline osu.Game/Beatmaps/IO/ArchiveReader.cs]: IDisposable without deconstructor, not sure if intended

File: osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                        handleEvents(beatmap, val);
+                        break;
+                    case Section.TimingPoints:
+                        handleTimingPoints(beatmap, val);
+                        break;
+                    case Section.Colours:
+                        handleColours(beatmap, key, val);
+                        break;
+                    case Section.HitObjects:
+                        beatmap.HitObjects.Add(HitObject.Parse(beatmap.Mode, val));
```
Comment (peppy) [inline osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs]: not too fond of having this the way it is. i'm fine to merge it for this PR, but can we agree on having something better like

```
var parser = HitObjectParser.ForMode(beatmap.Mode);

...
for (var line in file)
    parser.Parse(line);
```

**Comment by peppy** [comment]
🍦

**Pull request merged**